src/lib.rs

```diff
@@ -2069,6 +2069,51 @@ macro_rules! transmute_ref {
     }}
 }
 
+/// Includes a file and safely transmutes it to an arbitrary type.
+///
+/// The file will be included as a byte array, `[u8; N]`, which will be
+/// transmuted to another type, `T`. `include_bytes!("path")` returns a
+/// reference, i.e. `&T`. `T` is inferred from the calling context, and must
+/// implement [`FromBytes`].
+///
+/// The file is located relative to the current file (similarly to how modules
+/// are found). The provided path is interpreted in a platform-specific way at
+/// compile time. So, for instance, an invocation with a Windows path containing
+/// backslashes `\` would not compile correctly on Unix.
+///
+/// # Examples
+///
+/// Assume there are two files in the same directory with the following
+/// contents:
+///
+/// File 'data' (no trailing newline):
+///
+/// ```text
+/// abcd
+/// ```
+///
+/// File 'main.rs':
+///
+/// ```rust
+/// use zerocopy::include_bytes;
+/// # macro_rules! include_bytes {
+/// # ($file:expr) => { zerocopy::include_bytes!(concat!("../testdata/include_bytes/", $file)) };
+/// # }
+///
+/// fn main() {
+///     let as_u32: &u32 = include_bytes!("data");
+///     assert_eq!(as_u32, &u32::from_ne_bytes([b'a', b'b', b'c', b'd']));
+///     let as_i32: &i32 = include_bytes!("data");
+///     assert_eq!(as_i32, &i32::from_ne_bytes([b'a', b'b', b'c', b'd']));
+/// }
+/// ```
+#[macro_export]
+macro_rules! include_bytes {
+    ($file:expr $(,)?) => {
+        &$crate::transmute!(*::core::include_bytes!($file))
```

Comment:
Is the `$crate::transmute!` hygienic? The [blog post](https://gist.github.com/Kestrer/8c05ebd4e0e9347eb05f265dfb7252e1) you linked to doesn't specifically address `$crate` or naming macros by path.

Reply:
I believe so. That blog post says:
> Declarative macros provide a very useful feature, $crate, which is a special path root that unambiguously refers to the crate in which the declarative macro was written. 

The rust documentation supports this: https://doc.rust-lang.org/beta/reference/macros-by-example.html#hygiene

Reply:
I'm specifically worried about use in older editions. IIRC, it used to be the case that `foo::bar!()` wasn't valid - you just needed to do `#[macro_use] extern crate foo;` and then `bar!()` separately?

Reply:
Discussed offline; this is good as is.